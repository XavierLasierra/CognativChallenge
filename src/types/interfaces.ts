export interface IRecipe {
  _id: string;
  name: string;
  categoryId: string;
  categoryName: string;
  duration: number;
  complexity: string;
  favorite?: number;
  people: number;
  recommended?: number;
  ingredients: string;
  description: string;
  photo?: string;
}

export interface IRecipePreviewProps {
  recipe: IRecipe;
  actionOnPress: (recipe: string) => void;
}

export interface IRecipeSummaryProps {
  recipe: IRecipe;
  alignCenter?: boolean;
}

interface IParams {
  recipeId: string;
}
interface IRoute {
  params: IParams;
}
export interface IRecipeDetailsProps {
  route: IRoute;
  navigation: INavigation;
}

interface INavigation {
  push: (path: string, params: IParams) => void;
  pop: () => void;
}

export interface IHomeProps {
  navigation: INavigation;
}

export interface IRecipeProp {
  item: IRecipe;
}

export interface IBackButtonProps {
  actionOnPress: () => void;
}

interface IRecipeData {
  data: IRecipe[];
}

export interface IAxiosRecipeData {
  data: IRecipeData;
}
