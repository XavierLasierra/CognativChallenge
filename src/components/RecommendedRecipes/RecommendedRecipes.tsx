import React from "react";
import { Text, View, TouchableOpacity, Image, ScrollView } from "react-native";
import placeholder from "../../constants/placeholder.constants";
import { IRecipe, IRecommendedRecipesProps } from "../../types/interfaces";

import styles from "./recommendedRecipes.styles";

export default function RecommendedRecipes({
  recommendations,
  actionOnPress,
}: IRecommendedRecipesProps) {
  return (
    <View style={styles.recommendedContainer}>
      <Text style={styles.header}>Recommended</Text>
      <ScrollView
        horizontal={true}
        showsVerticalScrollIndicator={false}
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={styles.scroller}>
        {recommendations.map((item: IRecipe) => {
          return (
            <TouchableOpacity
              accessibilityRole={"button"}
              testID={`recommended-${item._id}`}
              onPress={() => actionOnPress(item._id)}
              key={item._id}>
              <View style={styles.recipeImageBox}>
                <Image
                  source={{ uri: item?.photo || placeholder.defaultImageUrl }}
                  style={styles.scrollerRecipe}
                />
              </View>
            </TouchableOpacity>
          );
        })}
      </ScrollView>
    </View>
  );
}
