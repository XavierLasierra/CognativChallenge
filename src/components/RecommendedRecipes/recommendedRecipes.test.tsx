import React from "react";
import { render, fireEvent, RenderAPI } from "@testing-library/react-native";
import { IRecipe } from "../../types/interfaces";

import RecommendedRecipes from "./RecommendedRecipes";

import recipesMock from "../../__mocks__/recipes.mock";

describe("Given a RecommendedRecipes component", () => {
  describe("When it is rendered", () => {
    let screen: RenderAPI;
    let actionOnPress: () => void;
    let recommendations: IRecipe[];
    beforeEach(() => {
      actionOnPress = jest.fn();
      recommendations = recipesMock.slice(0, 2);

      screen = render(
        <RecommendedRecipes
          recommendations={recommendations}
          actionOnPress={actionOnPress}
        />,
      );
    });

    test("Then should match the snapshot", () => {
      expect(screen).toMatchSnapshot();
    });

    describe("And you click on one of the recipies recommendations", () => {
      test("Then actionOnPress should have been called with the id of the clicked recipe", () => {
        const id = recommendations[0]._id;
        const button = screen.getByTestId(`recommended-${id}`);
        fireEvent(button, "press");

        expect(actionOnPress).toHaveBeenCalledWith(id);
      });
    });
  });
});
