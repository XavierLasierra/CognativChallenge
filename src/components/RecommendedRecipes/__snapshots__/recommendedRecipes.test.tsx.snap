// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a RecommendedRecipes component When it is rendered Then should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "justifyContent": "center",
      "marginBottom": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontWeight": "bold",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
        "textAlign": "left",
      }
    }
  >
    Recommended
  </Text>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingRight": 16,
      }
    }
    horizontal={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="recommended-52944"
      >
        <View
          style={
            Object {
              "height": 128,
              "width": 208,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.themealdb.com/images/media/meals/1520084413.jpg",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 128,
                "justifyContent": "center",
                "marginLeft": 16,
                "overflow": "hidden",
              }
            }
          />
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="recommended-52943"
      >
        <View
          style={
            Object {
              "height": 128,
              "width": 208,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://www.themealdb.com/images/media/meals/1520083578.jpg",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 4,
                "height": 128,
                "justifyContent": "center",
                "marginLeft": 16,
                "overflow": "hidden",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
